tests/TestCase/ORM/MarshallerTest.php

```diff
@@ -683,6 +685,46 @@ public function testBelongsToManyWithMixedJoinData()
         $this->assertEquals(1, $result->tags[1]->_joinData->active);
     }
 
+    public function testOneBelongsToManyWithNestedAssociations()
+    {
+        $this->tags->belongsToMany('Articles');
+        $data = [
+            'name' => 'new tag',
+            'articles' => [
+                // This nested article exists, and we want to update it.
+                [
+                    'id' => 1,
+                    'title' => 'New tagged article',
+                    'body' => 'New tagged article',
+                    'user' => [
+                        'id' => 1,
+                        'username' => 'newuser'
+                    ],
+                    'comments' => [
+                        ['comment' => 'New comment', 'user_id' => 1],
+                        ['comment' => 'Second comment', 'user_id' => 1],
+                    ]
+                ]
+            ]
+        ];
+        $marshaller = new Marshaller($this->tags);
+        $tag = $marshaller->one($data, ['associated' => ['Articles.Users', 'Articles.Comments']]);
+
+        $this->assertNotEmpty($tag->articles);
+        $this->assertCount(1, $tag->articles);
+        $this->assertInstanceOf('Cake\ORM\Entity', $tag->articles[0]);
```

Comment:
You should also assert that the article here is not new, meaning that it exists already in the database